09-Workflow-Management-with-targets.Rmd

```diff
@@ -0,0 +1,164 @@
+# Workflow Management with {targets}
+
+## Using the {targets} package to automate your analyses
+
+Data science workflows will often grow beyond a single script and can rapidly
+become unwieldy when many steps, datasets, and files are involved. This is both
+a challenge to keep track of (i.e., which scripts to run first, second, etc.)
+and a challenge to communicate to other users in the world of open science.
+
+Workflow management software is a solution to this that the ROSSyndicate has
+embraced. Specifically, we use the
+[{targets}](https://books.ropensci.org/targets/) R package. {targets} in
+particular allows you to use R to build out a pipeline (used interchangeably
+with workflow here) which the software automatically tracks so that it "knows"
+which steps have been run, which have failed, and which need to be re-run. Your
+entire analytical pipeline or workflow can be rerun with just the command
+`tar_make()` and you can visualize the connections between objects and
+functions in your pipeline with `tar_visnetwork()`.
+
+One standout strength of this workflow management architecture is that the
+software "knows" which downstream steps are affected by a change in the
+upstream code or datasets. This means that you as a human have no need to
+manually re-run scripts that are outdated by changes in code; {targets} knows
+which are affected, can tell you, and automatically re-run them.
+
+Not every analysis is a perfect fit for a {targets} workflow, however. Very
+small analyses may not be worth the time needed to invest to build out the
+structure of the pipeline. Workflows that are often re-run with updated
+datasets or which can perform many parallel iterations are especially good
+candidates for the {targets} pipeline style. Because the ROSSyndicate values
+literate code and {targets} does not work with Markdown or Quarto documents,
```

Comment:
Targets does work with these! Quarto seems to be buggy and bookdown is hard, though possible, to implement.

Reply:
Yes, yes. I should clarify that it doesn't render *and* track the output, aka it's a literate/visual output, but not compatible with the 'return' of values/environments, right? I'm trying to get at that aspect of {targets} that you learned about earlier this year.

Reply:
AH yes, basically. That's what I've managed to get {bookdown} to do, but it's sketchy